TTCN3_Project/javasrc/CANPortFilter/CANPortFilter.java

```diff
@@ -0,0 +1,84 @@
+package CANPortFilter;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+
+import org.etsi.ttcn.tri.TriAddress;
+import org.etsi.ttcn.tri.TriComponentId;
+import org.etsi.ttcn.tri.TriMessage;
+import org.etsi.ttcn.tri.TriPortId;
+import org.etsi.ttcn.tri.TriStatus;
+
+import com.testingtech.ttcn.tri.PortFilter;
+
+public class CANPortFilter extends PortFilter {
+	
+	File speedFile = new File("resource/Speed");
+	BufferedReader br = null;
+	String speed = null;
+
+	private static final long serialVersionUID = -3492721938799363917L;
+
+	/**
+	 * 
+	 * @param tsiPortId Identifies the system port.
+	 * @param SUTAddress Indicates an internal address of the SUT.
+	 * @param componentId Identifies the target component.
+	 * @param receivedMessage Contains the encoded value that is processed by the
+	 *        by the receive statement.
+	 */
+	public void triEnqueueMsg(TriPortId tsiPortId, TriAddress SUTAddress,
+			TriComponentId componentId, TriMessage receivedMessage) {
+		
+		// create a new file input stream
+		try {
+			br = new BufferedReader(new FileReader(speedFile));
+		} catch (FileNotFoundException e) {
+			System.err.println("Error! File " + speedFile.getAbsoluteFile() + " not found.");
+			e.printStackTrace();
+		}
+		
+		// read content from file input stream
+		try {			
+			speed = br.readLine();
+		} catch (IOException e) {
+			System.err.println("Error while reading file.");
+			e.printStackTrace();
+		}
+		
+		System.out.println("PortFilter " + this
+				+ " sends following outgoing message: "
+				+ speed);
+
+		receivedMessage.setEncodedMessage(speed.getBytes());
+		if (receivedMessage.getEncodedMessage().length > 0) { // do not enqueue
+																// zero length
+																// messages
+			super.triEnqueueMsg(tsiPortId, SUTAddress, componentId,
+					receivedMessage);			
+		}
+	}
+
+	/**
+	 * This method is called when the test case executes a send statement. 
+	 * 
+	 * @param componentId Identifies the sending component.
+	 * @param tsiPortId Identifies the system port.
+	 * @param address Target of the send statement.
+	 * @param sendMessage The message to be send.
+	 * 
+	 */
+	public TriStatus triSend(TriComponentId componentId, TriPortId tsiPortId,
+			TriAddress address, TriMessage sendMessage) {
+		System.out.println("PortFilter " + this
```

Comment:
"PortFilter receives following outgoing message:" -> These kind of strings should be in a separate constants class to provide better maintainability.